cli/module_generate/scripts/generate_stubs.go

```diff
@@ -0,0 +1,245 @@
+// Package scripts contains scripts that generate method stubs for modules
+package scripts
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"io"
+	"net/http"
+	"strings"
+	"text/template"
+	"unicode"
+
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/cli/module_generate/common"
+)
+
+//go:embed tmpl-module
+var goTmpl string
+
+// getClientCode grabs client.go code of component type.
+func getClientCode(module common.ModuleInputs) (string, error) {
+	url := fmt.Sprintf("https://raw.githubusercontent.com/viamrobotics/rdk/refs/tags/v%s/%ss/%s/client.go",
+		module.SDKVersion, module.ResourceType, module.ResourceSubtype)
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, url, nil)
+	if err != nil {
+		return "", errors.Wrapf(err, "cannot get client code")
+	}
+
+	//nolint:bodyclose
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", errors.Wrapf(err, "cannot get client code")
+	}
+	defer utils.UncheckedErrorFunc(resp.Body.Close)
+	if resp.StatusCode != http.StatusOK {
+		return "", errors.Errorf("unexpected http GET status: %s getting %s", resp.Status, url)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return url, errors.Wrapf(err, "error reading response body")
+	}
+	clientCode := string(body)
+	return clientCode, nil
+}
+
+// setGoModuleTemplate sets the imports and functions for the go method stubs.
+func setGoModuleTemplate(clientCode string, module common.ModuleInputs) (*common.GoModuleTmpl, error) {
+	var goTmplInputs common.GoModuleTmpl
+
+	if module.ResourceSubtype == "input" {
+		module.ResourceSubtypePascal = "Controller"
+	}
+
+	fset := token.NewFileSet()
+	node, err := parser.ParseFile(fset, "", clientCode, parser.AllErrors)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse client code")
+	}
+
+	var imports []string
+	for _, imp := range node.Imports {
+		path := imp.Path.Value
+		// check for the specific import path and set the alias
+		if path == `"go.viam.com/rdk/vision"` {
+			imp.Name = &ast.Ident{Name: "vis"}
+		}
+		if imp.Name != nil {
+			path = fmt.Sprintf("%s %s", imp.Name.Name, path)
+		}
+		imports = append(imports, path)
+	}
+	var functions []string
+	ast.Inspect(node, func(n ast.Node) bool {
+		if funcDecl, ok := n.(*ast.FuncDecl); ok {
+			name, args, returns := parseFunctionSignature(module.ResourceSubtype, module.ResourceSubtypePascal, funcDecl)
+			if name != "" {
+				functions = append(functions, formatEmptyFunction(module.ModuleCamel+module.ModelPascal, name, args, returns))
+			}
+		}
+		return true
+	})
+
+	// add DoCommand function stub to mlmodel
+	if module.ResourceSubtype == "mlmodel" {
+		doCommandFunction := formatEmptyFunction(module.ModuleCamel+module.ModelPascal,
+			"DoCommand",
+			"ctx context.Context, cmd map[string]interface{}",
+			[]string{"map[string]interface{}", "error"})
+		functions = append(functions, doCommandFunction)
+	}
```

Comment:
curious why we're doing this? `DoCommand` doesn't exist in the mlmodel proto iirc, so I'd be a bit surprised if calling `doCommand`on a mlmodel client would work.

Reply:
yeah good question. so mlmodel actually _does_ implement doCommand but it is not in the client code. You can see [here](https://docs.viam.com/services/ml/#api) that it does have `DoCommand` but it is not obvious since it is not in the client code. Naveed found that is being called from [here](https://github.com/viamrobotics/rdk/blob/2356447a29bb90b4c07a0eae480faaf19a42224a/resource/resource.go#L278) however I still do not understand how the client code works without having a `DoCommand`. I told Bijan and Khari about this so they are aware that it is missing, but for now in order to move on without errors I had to manually add in a `DoCommand`. Let me know if that doesn't make sense